widget/embedded/src/pages/SettingsPage.tsx

```diff
@@ -1,93 +1,238 @@
+});
+
+const Head = styled('div', {
+  display: 'flex',
+  justifyContent: 'start',
+  alignItems: 'center',
+  marginBottom: '10px',
+});
+
+interface SettingItemPropsTypes {
+  title: string;
+}
+
+function SettingItemTitle({ title }: SettingItemPropsTypes) {
+  return (
+    <Typography variant="title" size="xmedium" color="neutral900">
+      {i18n.t(title)}
+    </Typography>
+  );
+}
+
 interface PropTypes {
   supportedSwappers?: string[];
   singleTheme?: boolean;
 }
+
 export function SettingsPage({ supportedSwappers, singleTheme }: PropTypes) {
   const slippage = useSettingsStore.use.slippage();
   const setSlippage = useSettingsStore.use.setSlippage();
-  const disabledLiquiditySources =
-    useSettingsStore.use.disabledLiquiditySources();
   const customSlippage = useSettingsStore.use.customSlippage();
   const setCustomSlippage = useSettingsStore.use.setCustomSlippage();
-  const theme = useSettingsStore.use.theme();
-  const setTheme = useSettingsStore.use.setTheme();
-  const swappers = useMetaStore.use.meta().swappers;
+
   const navigate = useNavigate();
   const { navigateBackFrom } = useNavigateBack();
 
   const infiniteApprove = useSettingsStore.use.infiniteApprove();
   const toggleInfiniteApprove = useSettingsStore.use.toggleInfiniteApprove();
   const loadingMetaStatus = useMetaStore.use.loadingStatus();
 
-  const uniqueSwappersGroups: Array<{
-    title: string;
-    logo: string;
-    type: 'BRIDGE' | 'AGGREGATOR' | 'DEX';
-    selected: boolean;
-  }> = [];
-  removeDuplicateFrom(swappers.map((s) => s.swapperGroup))
-    .map((swapperGroup) => {
-      return swappers.find((s) => s.swapperGroup === swapperGroup);
-    })
-    .find((s) => {
-      if (s) {
-        for (const type of s.types) {
-          uniqueSwappersGroups.push({
-            title: s.swapperGroup,
-            logo: s.logo,
-            type,
-            selected: !disabledLiquiditySources.includes(s.swapperGroup),
-          });
-        }
-      }
-    });
-  supportedSwappers &&
-    uniqueSwappersGroups.filter(
-      (item) => supportedSwappers.filter((s) => s === item.title).length > 0
-    );
-
-  const supportedUniqueSwappersGroups = supportedSwappers
-    ? uniqueSwappersGroups.filter(
-        (item) => supportedSwappers.filter((s) => s === item.title).length > 0
-      )
-    : uniqueSwappersGroups;
+  const supportedUniqueSwappersGroups =
+    getUniqueSwappersGroups(supportedSwappers);
+
+  const bridgeSources = supportedUniqueSwappersGroups.filter(
+    (uniqueItem) =>
+      uniqueItem.type === 'BRIDGE' || uniqueItem.type === 'AGGREGATOR'
+  );
+
+  const selectedBridgeSources = bridgeSources.filter(
+    (uniqueItem) => uniqueItem.selected
+  );
+
+  const exchangeSources = supportedUniqueSwappersGroups.filter(
+    (uniqueItem) => uniqueItem.type === 'DEX'
+  );
+
+  const selectedExchangeSources = exchangeSources.filter(
+    (uniqueItem) => uniqueItem.selected
+  );
+
+  const bridgeItem = {
+    id: '1',
```

Comment:
please use a meaningful name instead of number, since we reorder the the list, we should update the number always. you can use something like `bridge-item`.